README updates: link to Community page, drop obsolete info, etc

chalin opened the pull request:
cc @thisisnotapril @srini100

srini100 reviewed (approved):
LGTM. One small concern I have is that the link to the governance doc is now buried deeper.

chalin commented:
> LGTM

Thanks for the review.

> One small concern I have is that the link to the governance doc is now buried deeper.

I've added back direct links. PTAL. 

FYI: My original reasoning was that folks generally don't visit the README page directly, but rather the grpc-community homepage, in which case they can see links to governance doc in the repo listing. Anyhow, I've added the links back.

chalin commented on README.md:
```diff
@@ -1,25 +1,14 @@
-repositories of community contributions around gRPC. If you are interested, please see
-details [here](https://github.com/grpc/grpc-community/blob/master/grpc_ecosystem.md).
-
-# Known useful contributions around github
-
-- [gRPC gateway](https://github.com/gengo/grpc-gateway)
-- [Polyglot](https://github.com/grpc-ecosystem/polyglot)
-- [Zipkin Support for gRPC](https://github.com/openzipkin/brave/tree/master/brave-grpc)
-- [Prometheus monitoring for gRPC Go](https://github.com/mwitkow/go-grpc-prometheus)
-- [Flatbuffer support for gRPC C++](https://github.com/google/flatbuffers/tree/master/grpc)
-- [Go Kit with gRPC as transport](https://github.com/go-kit/kit/tree/master/transport/grpc)
-- [Stackdriver Trace Support for gRPC Go](https://godoc.org/cloud.google.com/go/trace)
-- [grpc-java-contrib](https://github.com/salesforce/grpc-java-contrib)
-- [grpc-errors](https://github.com/avinassh/grpc-errors)
```
I removed these because several are outdated, and besides, https://github.com/grpc-ecosystem/awesome-grpc serves the purpose of listing awesome community projects :).

thisisnotapril commented:
LGTM; thank you Patrice!
